tests/ignite/handlers/test_time_limit.py

```diff
@@ -0,0 +1,31 @@
+import time
+
+import pytest
+
+from ignite.engine import Engine, Events
+from ignite.handlers import TimeLimit
+
+
+def test_arg_validation():
+
+    with pytest.raises(ValueError, match=r"Argument limit_sec should be a positive integer."):
+        TimeLimit(limit_sec=-5)
+
+    with pytest.raises(TypeError, match=r"Argument limit_sec should be an integer."):
+        TimeLimit(limit_sec="abc")
+
+
+def test_terminate_on_time_limit():
+    def _train_func(engine, batch):
+        time.sleep(1)
+
+    def _test(n_iters, limit):
+        started = time.time()
+        trainer = Engine(_train_func)
+        trainer.add_event_handler(Events.ITERATION_COMPLETED, TimeLimit(limit))
+        trainer.run(range(n_iters))
+        elapsed = round(time.time() - started)
+        assert elapsed <= limit + 1
```

Comment:
Please add the following assertion 
```suggestion
        assert elapsed <= limit + 1
        assert trainer.state.iteration == min(n_iters, limit)
```

Reply:
I'm not sure about checking iteration vs time limit in general...

Reply:
I agree that this may not be appropriate in general situations but it would be surprising if it does not work here. Feel free to ignore this assertion if not relevant.